ath79:add support for COMFAST CF-E5/E7

fall513 opened the pull request:
COMFAST CF-E5/E7 is a outdoor 4G LTE AP with PoE support, based on
Qualcomm/Atheros QCA9531.

Short specification:

    2x 10/100 Mbps Ethernet, with 24v PoE support
    64 MB of RAM (DDR2)
    16 MB of FLASH (SPI)
    2T2R 2.4 GHz, 802.11b/g/n
    built-in 1x 3 dBi antennas
    output power (max): 80 mW (19 dBm)
    Qucetel EC20 LTE MODULE(1x external detachable antenna)

Flash instruction:

Original firmware is based on OpenWrt.
Use sysupgrade image directly in vendor GUI.

Signed-off-by: Ding Tengfei <dtf@comfast.cn>

ynezz commented on target/linux/ath79/dts/qca9531_comfast_cf-e5.dts:
```diff
@@ -0,0 +1,175 @@
+				read-only;
+			};
+
+			art: partition@1 {
```
@fall513 It's `partition@addresss` so it should be written as `partition@10000`

fall513 commented:
Fixed the partition address. Thanks @ynezz

981213 reviewed (changes requested):
Several things inline :)

fall513 commented:
@981213

Thanks for your review. I've fixed the issues and pushed again.

ynezz commented:
@fall513 You need to do rebase `git pull --rebase && git push -f` in order to avoid the `Merge branch 'master' into ath79-add-support-for-COMFAST-CF-E5/E7` commits. Please fix it,  so you've only `ath79:add support for COMFAST CF-E5/E7 ` in your branch and this PR.

ynezz commented:
@fall513 Looks better now, but now your commit is missing Subject line `ath79: add support for COMFAST CF-E5/E7`

ynezz commented on target/linux/ath79/dts/qca9531_comfast_cf-e5.dts:
```diff
@@ -0,0 +1,170 @@
+		};
+	};
+
+	gpio-export {
```
@fall513 We're trying to get rid of the `gpio-export` functionality as it's hack for missing kernel functionality, which was rejected in upstream kernel long time ago, for details see this email http://lists.infradead.org/pipermail/openwrt-devel/2019-February/015772.html, discussion in #1366 or https://github.com/openwrt/openwrt/pull/1814#issuecomment-462942022.

So I'm wondering if it would be possible to replace the `gpio-export` construct with the userspace gpio-switch alternative, for example https://github.com/openwrt/openwrt/blob/master/target/linux/layerscape/base-files/etc/board.d/03_gpio_switches#L20-L22

fall513 replied:
OK, I will modify and do some test.

ynezz commented on target/linux/ath79/dts/qca9531_comfast_cf-e5.dts:
```diff
@@ -0,0 +1,170 @@
+
+		poll-interval = <20>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&jtag_disable_pins &led_wan_pin>;
```
@fall513 You're not using GPIOs 0-3 in `keys` node, but you're using them for LEDs, so please move the pinctrl under `leds` node.

ynezz commented on target/linux/ath79/dts/qca9531_comfast_cf-e5.dts:
```diff
@@ -0,0 +1,170 @@
+			};
+
+			partition@ff0000 {
+				label = "art-bakup";
```
@fall513 `art-bakup` -> `art-backup`

ynezz commented on target/linux/ath79/dts/qca9531_comfast_cf-e5.dts:
```diff
@@ -0,0 +1,170 @@
+};
+
+&pinmux {
+	led_wan_pin: pinmux_wan_pin {
```
@fall513  `pinmux_wan_pin` -> `pinmux_led_wan_pin`

fall513 commented:
@ynezz 
Thanks for your review. I've fixed the issues and pushed again.

ynezz commented:
@fall513 Can you please rebase, I'm going to merge this soon.

cfpublish committed:
ath79:add support for COMFAST CF-E5/E7

COMFAST CF-E5/E7 is a outdoor 4G LTE AP with PoE support, based on
Qualcomm/Atheros QCA9531.

Short specification:

    2x 10/100 Mbps Ethernet, with 24v PoE support
    64 MB of RAM (DDR2)
    16 MB of FLASH (SPI)
    2T2R 2.4 GHz, 802.11b/g/n
    built-in 1x 3 dBi antennas
    output power (max): 80 mW (19 dBm)
    Qucetel EC20 LTE MODULE(1x external detachable antenna)

Flash instruction:

Original firmware is based on OpenWrt.
Use sysupgrade image directly in vendor GUI.

Signed-off-by: Ding Tengfei <dtf@comfast.cn>

fall513 commented:
@ynezz I've fixed the issues and pushed again.

ynezz commented:
Thanks, I've just merged it to my [staging tree](https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=shortlog;h=refs/heads/staging)

adrianschmutzler commented:
Although this PR has been has been closed already, I would be interested in some additional data about the device(s) in case you still have access to it:

I'm currently working on a PR to store the label-mac-address in the DTS (https://github.com/openwrt/openwrt/pull/2159).

I would appreciate if you could provide information about your device:
Please run
`find /proc/device-tree/ -name "*mac-address*"`
on your device (just with the current firmware, no need to update anything; just has to be ath79, no ar71xx).
For each of the returned paths (if there are any), retrieve the mac-address with `cat`, e.g.
`cat /proc/device-tree/ahb/eth@19000000/mac-address`

Now check which of those returned MAC addresses matches the one on the label/sticker/case of your device and tell me the device-tree path leading to it. Note that you do not have to post your MAC addresses here, but just a statement about which match, e.g. like
```
/proc/device-tree/ahb/eth@1a000000/mac-address
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address-increment
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address
/proc/device-tree/ahb/eth@19000000/mac-address <== this one
/proc/device-tree/ahb/eth@19000000/mtd-mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mtd-mac-address
```

There may be one, two or no paths giving the correct address. It may also happen that the `find` command returns nothing. Thanks.

fall513 commented:
root@OpenWrt:/sys/firmware/devicetree/base# find /proc/device-tree/ -name "*mac-
address*" 
/proc/device-tree/ahb/eth@1a000000/mac-address
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address
/proc/device-tree/ahb/eth@19000000/mac-address      <== this one
/proc/device-tree/ahb/eth@19000000/mtd-mac-address





------------------ 原始邮件 ------------------
发件人: "Adrian Schmutzler"<notifications@github.com>;
发送时间: 2019年7月5日(星期五) 晚上6:59
收件人: "openwrt/openwrt"<openwrt@noreply.github.com>;
抄送: "147207374"<147207374@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [openwrt/openwrt] ath79:add support for COMFAST CF-E5/E7 (#1759)




Although this PR has been has been closed already, I would be interested in some additional data about the device(s) in case you still have access to it:
 
I'm currently working on a PR to store the label-mac-address in the DTS (#2159).
 
I would appreciate if you could provide information about your device:
 Please run
 find /proc/device-tree/ -name "*mac-address*"
 on your device (just with the current firmware, no need to update anything; just has to be ath79, no ar71xx).
 For each of the returned paths (if there are any), retrieve the mac-address with cat, e.g.
 cat /proc/device-tree/ahb/eth@19000000/mac-address
 
Now check which of those returned MAC addresses matches the one on the label/sticker/case of your device and tell me the device-tree path leading to it. Note that you do not have to post your MAC addresses here, but just a statement about which match, e.g. like
 /proc/device-tree/ahb/eth@1a000000/mac-address /proc/device-tree/ahb/eth@1a000000/mtd-mac-address-increment /proc/device-tree/ahb/eth@1a000000/mtd-mac-address /proc/device-tree/ahb/eth@19000000/mac-address <== this one /proc/device-tree/ahb/eth@19000000/mtd-mac-address /proc/device-tree/ahb/apb/wmac@18100000/mac-address /proc/device-tree/ahb/apb/wmac@18100000/mtd-mac-address  
There may be one, two or no paths giving the correct address. It may also happen that the find command returns nothing. Thanks.
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

adrianschmutzler commented:
Thanks, so this one will be
`label-mac-device = &eth0`

fall513 commented:
Correct.


root@OpenWrt:/sys/firmware/devicetree/base# hexdump /proc/device-tree/ahb/eth@19
000000/mac-address
0000000 4045 4545 7744 





eth0      Link encap:Ethernet  HWaddr 40:45:45:45:77:44  
          inet addr:192.192.0.106  Bcast:192.192.0.255  Mask:255.255.255.0
          inet6 addr: fe80::4245:45ff:fe45:7744/64 Scope:Link
          inet6 addr: fd16:c76e:8f6a::878/128 Scope:Global
          inet6 addr: fd16:c76e:8f6a:0:4245:45ff:fe45:7744/64 Scope:Global
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:288 errors:0 dropped:0 overruns:0 frame:0
          TX packets:68 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:70442 (68.7 KiB)  TX bytes:6871 (6.7 KiB)
          Interrupt:4 



------------------ 原始邮件 ------------------
发件人: "Adrian Schmutzler"<notifications@github.com>;
发送时间: 2019年7月5日(星期五) 晚上7:24
收件人: "openwrt/openwrt"<openwrt@noreply.github.com>;
抄送: "147207374"<147207374@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [openwrt/openwrt] ath79:add support for COMFAST CF-E5/E7 (#1759)




Thanks, so this one will be
 label-mac-device = &eth0
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

grosjo commented:
@fall513 : I just installed the firmware but the vendor keeps the old root password. Any clue how to get rid of the old root password (after installing the new firmware sadly) ?

grosjo commented:
Ok, found it, the "reset" is 7 sec

grosjo commented:
Actually , the installation method desciption is totally wrong, as the vendor passwrd is kept in the vendor gui.

THe process would /not/ be using vendor GUi, but 
(a) pushing the device reset pushing
(b) plugin power for exactly 7 sec
(c) connecte to 192.168.1.1 to update the firmware


How to update the commit ?

fall513 commented:
I think you can press reset button to restore default configs after upgraded.
Press reset button when booting is another way to upgrade, but some system have network card's issue, such as windows 10



---Original---
From: "Joan Moreau"<notifications@github.com&gt;
Date: Sat, Aug 31, 2019 21:20 PM
To: "openwrt/openwrt"<openwrt@noreply.github.com&gt;;
Cc: "Mention"<mention@noreply.github.com&gt;;"Ding Tengfei"<147207374@qq.com&gt;;
Subject: Re: [openwrt/openwrt] ath79:add support for COMFAST CF-E5/E7 (#1759)



Actually , the installation method desciption is totally wrong, as the vendor passwrd is kept in the vendor gui.
 
THe process would /not/ be using vendor GUi, but
 (a) pushing the device reset pushing
 (b) plugin power for exactly 7 sec
 (c) connecte to 192.168.1.1 to update the firmware
 
How to update the commit ?
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.
